### PR TITLE
Engine cleanup

### PR DESCRIPTION
**Pull request by mugling**
Some initial cleanup to keep down line count of a big reworking of vehicle engines that will follow.

First two commits are migrations and linting. Third deprecates the useless plasma engine and the associated hydrogen tank. Once we have generic engine fuels  via JSON we can add some more interesting alternatives then.

### CODE REVIEW

**Comment by BorkBorkGoesTheCode** [comment]
Why can't you make the plasma engine useful instead of dropping it?

**Comment by Coolthulhu** [comment]
Because it uses a very rare resource that everyone wants for diamond weapons.
It's a money-fueled furnace.

**Comment by mugling** [comment]
> Because it uses a very rare resource that everyone wants for diamond weapons.
> It's a money-fueled furnace.

It's entirely non salvageable in it's current form. Once we have generic fuel we can add lots more novel items

File: src/savegame_json.cpp [hunk 1]
```diff
@@ -1665,7 +1666,9 @@ void vehicle_part::deserialize(JsonIn &jsin)
         { "external_water_tank", { "external_tank", "water_clean" } },
         { "water_tank_barrel", { "tank_barrel", "water_clean" } },
 
-        { "napalm_tank", { "tank", "napalm" } }
+        { "napalm_tank", { "tank", "napalm" } },
+
+        { "hydrogen_tank", { "tank", "plasma" } }
```
Comment (mugling) [inline src/savegame_json.cpp]: Lets go with `none` then. The odds of anyone having plasma in a tank are defacto zero given how exceedingly rare it is and the other uses being much more valuable (plasma weapons and cvd machines)

**Comment by SeanMirrsen** [comment]
>It's entirely non salvageable in it's current form. Once we have generic fuel we can add lots more novel items

Proper steam boilers? Stirling engines? Wind-up mechanics? Wind power? :P

**Pull request merged**